### ISSUE TITLE
zcash_client_sqlite: Inconsistency between tree size & checkpoint position updates.

### ISSUE THREAD

**Issue by nuttycom** [zcash/librustzcash#1302]
This selection of mainnet checkpoint data illustrates the issue: 
```
height  | s_tree_sz | position | o_tree_sz | position
--------|-----------|----------|-----------|---------
2399872 | 73205447  | 73205446 | 48705967  | 48705960
2399871 | 73205445  |          | 48705967  |  
2399870 | 73205445  |          | 48705967  |  
2399869 | 73205445  |          | 48705967  |  
2399868 | 73205445  | 73205444 | 48705967  | 48705960
2399867 | 73205443  |          | 48705967  |  
2399866 | 73205443  |          | 48705967  |  
2399865 | 73205443  | 73205442 | 48705967  | 48705960 <-- problem here
2399864 | 73205441  | 73205440 | 48705961  | 48705960
2399863 | 73205439  |          | 48705961  |  
2399862 | 73205439  | 73205438 | 48705961  | 48705960
2399861 | 73205437  | 73205436 | 48705959  | 48705958
2399860 | 73205437  | 73205436 | 48705957  | 48705956
2399859 | 73205437  | 73205436 | 48705955  | 48705954
```
In this trace, it appears that an Orchard checkpoint is inserted at height `2399865` that does not reflect the additions to the Orchard note commitment tree that appear in that block. None of the later checkpoints then observe that adjustment.

**Comment by nuttycom** [comment]
Fixed by #1303

**Comment by nuttycom** [comment]
This does not reproduce when using a batch size of 10,000 blocks, but with a batch size of 100 blocks it reproduces reliably.
